drivers/virtualbox/virtualbox.go

```diff
@@ -62,17 +65,24 @@ type Driver struct {
 // NewDriver creates a new VirtualBox driver with default settings.
 func NewDriver(hostName, storePath string) *Driver {
 	return &Driver{
-		VBoxManager: NewVBoxManager(),
-		BaseDriver: &drivers.BaseDriver{
-			MachineName: hostName,
-			StorePath:   storePath,
-		},
+		VBoxManager:         NewVBoxManager(),
```

Comment:
Non-exported fields are not marshaled
Le mer. 13 janv. 2016 à 22:08, Nathan LeClaire <notifications@github.com> a
écrit :

> In drivers/virtualbox/virtualbox.go
> <https://github.com/docker/machine/pull/2806#discussion_r49650501>:
>
> > @@ -62,17 +65,24 @@ type Driver struct {
> >  // NewDriver creates a new VirtualBox driver with default settings.
> >  func NewDriver(hostName, storePath string) *Driver {
> >  	return &Driver{
> > -		VBoxManager: NewVBoxManager(),
> > -		BaseDriver: &drivers.BaseDriver{
> > -			MachineName: hostName,
> > -			StorePath:   storePath,
> > -		},
> > +		VBoxManager:         NewVBoxManager(),
>
> Will unmarshaling hosts from old versions work OK with these new
> additions? I think it might be OK, because the newly added objects are not
> pointers (consequently they will be initialized to an empty struct, not
> nil), but we should make sure to test thoroughly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/machine/pull/2806/files#r49650501>.
>